test/functional/test_framework/crypto/chacha20.py

```diff
@@ -62,6 +64,34 @@ def chacha20_block(key, nonce, cnt):
     # Produce byte output
     return b''.join(state[i].to_bytes(4, 'little') for i in range(16))
 
+class FSChaCha20:
+    """Rekeying wrapper stream cipher around ChaCha20."""
+    def __init__(self, initial_key, rekey_interval=REKEY_INTERVAL):
+        self._key = initial_key
+        self.rekey_interval = rekey_interval
+        self.block_counter = 0
+        self.chunk_counter = 0
+        self.keystream = b''
```

Comment:
nit: I think all member variables are not meant to be accessed from the outside and could hence be prefixed with `_` (same for the `get_key_stream_bytes` method).
```suggestion
        self._key = initial_key
        self._rekey_interval = rekey_interval
        self._block_counter = 0
        self._chunk_counter = 0
        self._keystream = b''
```

Reply:
are private functions prefixed with single `_` too? i think i've only come across private functions prefixed with double `_`.

Reply:
> are private functions prefixed with single _ too?

I think so, at least that's what we also do for some tests and in the framework, see e.g ```$ git grep "def _[a-zA-Z]" ./test``` versus ```$ git grep "def __[a-zA-Z]" ./test``` (the latter yields results in the `__foobar__` form for special operations like declaring ctors, dtors, operator overloading etc.). Using two underscores seems to be more than only a convention, triggering name mangling: https://stackoverflow.com/a/1301369